caom2pipe/manage_composable.py

```diff
@@ -2924,3 +2925,146 @@ def find_missing(compare_this, to_this):
     :return: list of elements from to_this not in compare_this
     """
     return list(set(compare_this).difference(to_this))
+
+
+class ValueRepairCache(Cache):
+    """Use a cache file to repair metadata values.
+
+    The cache file is identified in the config.yml file.
+
+    The cache file contents should look like:
+    value_repair:  # must have this value
+      observation.type:  # a caom2 entity.attribute value
+        dark: DARK       # can have multiple entries per attribute, 'any'
+                         # will replace all possible values, can have more
+                         # than one replacement value
+
+    Caveats:
+
+    Some CAOM2 attributes are not assignable. These attributes cannot be
+    repaired using this class.
+
+    List content cannot be repaired using this class.
+
+    Use the string value 'none' to set an attribute to None.
+
+    Use '*' to replace part of a string only works with str values, and only
+    trailing changes, because there haven't been any preceding cases that I've
+    needed to write code for yet.
```

Comment:
Another limitation is that it's not working on attributes of a specific component (plane or artifact). It's all the attributes matching that name in all components.

Reply:
The asterisk only applies to replacement values, not data member identifiers. Data member identifiers have to be complete, and a correct python variable. I was going to use the `ObsBlueprint` lookup values as keys in the config file, but that increased the amount of code, and that code seemed brittle as well. So, then I decided to just rely on python's help for naming.

Configurations will possibly break with every release of caom2.4, but I think it's better to change a file on disk, than to have to rebuild a Docker image, to get correctness. I can be argued out of this position.

Reply:
The comment was not about the '*' but about the documentation of limitations. Just wanted to point out that the updates do not consider component identifiers and apply to all the matches in the observation. If that's worth pointing out.

Reply:
Value repair should ONLY be working on the attributes of a specific component. The key names in the cache are fully-qualified to ensure that. If there's something in the documentation or implementation that implies/executes otherwise, it's very wrong.

What did I miss?